lib/pf/vlan/pool.pm

```diff
@@ -0,0 +1,154 @@
+package pf::vlan::pool;
+
+=head1 NAME
+
+pf::vlan::pool - Object oriented module for VLAN isolation oriented functions
+
+=head1 SYNOPSIS
+
+The pf::vlan::pool module contains the functions necessary for the VLAN Pool.
+
+=cut
+
+use strict;
+use warnings;
+
+use Log::Log4perl;
+
+use pf::config;
+use pf::util;
+use pf::log();
+
+use pf::node;
+
+use Number::Range;
+
+=head1 SUBROUTINES
+
+Warning: The list of subroutine is incomplete
+
+=cut
+
+=head2 new
+
+Constructor.
+
+=cut
+
+sub new {
+    my $logger =  pf::log::get_logger();
+    $logger->debug("instantiating new pf::vlan::pool object");
+    my ( $class, %argv ) = @_;
+    my $this = bless {}, $class;
+    return $this;
+}
+
+sub getVlanPool {
+    my ($self, $vlan, $switch, $ifIndex, $mac, $node_info, $connection_type, $user_name, $ssid, $radius_request, $role) = @_;
+    my $logger =  pf::log::get_logger();
+
+    return $vlan if $self->rangeValidator($vlan);
+    my $range = Number::Range->new($vlan);
+
+    $vlan = $self->getRoundRobin($mac, $role, $range);
+    return $vlan;
+}
+
+=head2 rangeValidator
+
+Validate the range definition
+Should be something like that 20..23 or 20..23,27..30
+
+=cut
+
+sub rangeValidator {
+    my ($self, $range) =@_;
+    my $rangesep = qr/(?:\.\.)/;
+    my $sectsep  = qr/(?:\s|,)/;
+    my $validation = qr/(?:
+         [^0-9,. -]|
+         $rangesep$sectsep|
+         $sectsep$rangesep|
+         \d-\d|
+         ^$sectsep|
+         ^$rangesep|
+         $sectsep$|
+         $rangesep$
+         )/x;
+    return 1 if ($range =~ m/$validation/g);
+    return 0;
+}
+
+
+=head2 getRoundRobin
+
+Return the vlan id based on round robin
+
+=cut
+
+sub getRoundRobin {
+    my ($self, $mac, $role, $range) = @_;
+    my $logger =  pf::log::get_logger();
+
+    my $vlan_count = $range->size;
+    my $node_info_complete = node_view($mac);
+    if ( defined($node_info_complete->{'last_vlan'}) && $range->inrange($node_info_complete->{'last_vlan'}) ) {
+        $logger->debug("NODE LAST VLAN ".$node_info_complete->{'last_vlan'});
+        return ($node_info_complete->{'last_vlan'});
+    }
+    my $last_reg_mac = node_last_reg_non_inline_on_category($mac, $role);
```

Comment:
The last reg non inline on category value must be shared between all the radius process.
The better way is to take it from the database